ext/standard/password.c

```diff
@@ -61,7 +81,14 @@ static php_password_algo php_password_determine_algo(const char *hash, const siz
 {
 	if (len > 3 && hash[0] == '$' && hash[1] == '2' && hash[2] == 'y' && len == 60) {
 		return PHP_PASSWORD_BCRYPT;
+	} 
+#if HAVE_ARGON2LIB
+	if (hash[0] == '$' && strstr(hash, "argon2i")) {
```

Comment:
This will check whether argon2i is contained anywhere in the string. Maybe doing something more precise like `memcmp(hash + 1, "argon2i", sizeof("argon2i") - 1)`, with appropriate length check beforehand, would be preferable?